restore-hooks_product-requirements.md

```diff
@@ -0,0 +1,376 @@
+
+**<span style="text-decoration:underline;">______________________________________________________________</span>**
+
+**<span style="text-decoration:underline;">USE CASE 5</span>**
+
+
+    **Title: **Check for latest snapshot 
+
+
+    **Description: **As a user, I would like Velero to run a check for the latest snapshot in object storage prior to starting restore operations on a pod. 
+
+**<span style="text-decoration:underline;">______________________________________________________________</span>**
+
+**<span style="text-decoration:underline;">USE CASE 6</span>**
+
+
+    **Title: **Specify backup snapshot to use for restore
+
+
+    **Description: **As a user, I would like the option to specify a specific backup snapshot for use by Velero during restore create. I would like to do this instead of using the default most recent backup restore snapshot.
+
+**<span style="text-decoration:underline;">______________________________________________________________</span>**
+
+**<span style="text-decoration:underline;">USE CASE 7</span>**
+
+
+    **Title: **Include or exclude namespaces from restore
+
+
+    **Description: **As a user, I would like to include all resources in namespaces contained in a backup  by default in my restore spec. 
+
+
+    If velero is asked to restore something that already exists in a pod, the restore will not return a success but will still work - Ashish needs to verify.
+
+**<span style="text-decoration:underline;">______________________________________________________________</span>**
+
+**<span style="text-decoration:underline;">USE CASE 8</span>**
+
+
+    **Title: **Restore of a stateful application (Unquiescing data from a quiesced backup)
+
+
+    **Description: **As a user, I would like to prepare a backup to be restored without having to freeze an in-use volume/volume that an application is using.
+
+
+    As a user, I would like to unquiesce data during a restore to prevent the need to shut down the database and disrupt the application end user experience. 
+
+
+    _Note: Some dbs have built in way to back up _
+
+**<span style="text-decoration:underline;">______________________________________________________________</span>**
+
+**<span style="text-decoration:underline;">USE CASE 9</span>**
+
+
+    **Title: **Display/surface restore status 
+
+
+    **Description:** As a user, I would like to see the status of my restore status surfaced from the pod volume restore status. Including statuses:  Pending, Running/In Progress, Succeeded, Failed, Unknown.
+
+**<span style="text-decoration:underline;">______________________________________________________________</span>**
+
+**<span style="text-decoration:underline;">USE CASE 10</span>**
+
+
+    **Title: **Restore metadata
+
+
+    **Description: **As a user, I would like to have the metadata of the contents of what was restored using Velero.
```

Comment:
Are the contents referred to here the contents of what the hook restored? For example, this might be database files it reloaded, copied, or otherwise manipulated.

Reply:
@nrb yes that's the intent of this use case. Do you feel anything else needs to be included here to make this more clear?

Reply:
I don't think so; I will have to think about how we will have to provide a place for hooks to report what they did, though. Right now, there's no standard place for hooks, either backup or restore, to place data into Velero.

Reply:
Agree on this. And I think usecase 8, above, covered/should cover reporting status of the restore hook.
One way to surface the output of a restore hook would be to add that data as an annotation to the pod.